aea/helpers/search/models.py

```diff
@@ -840,6 +1383,71 @@ def __str__(self):
             self.attribute_name, self.constraint_type
         )
 
+    def encode(self) -> models_pb2.Query.ConstraintExpr.Constraint:
+        """
+        Encode an instance of this class into a protocol buffer object.
+
+        :return: the matching protocol buffer object
+        """
+        constraint = models_pb2.Query.ConstraintExpr.Constraint()
+        constraint.attribute_name = self.attribute_name
+
+        if (
+            self.constraint_type.type == ConstraintTypes.EQUAL
+            or self.constraint_type.type == ConstraintTypes.NOT_EQUAL
+            or self.constraint_type.type == ConstraintTypes.LESS_THAN
+            or self.constraint_type.type == ConstraintTypes.LESS_THAN_EQ
+            or self.constraint_type.type == ConstraintTypes.GREATER_THAN
+            or self.constraint_type.type == ConstraintTypes.GREATER_THAN_EQ
+        ):
+            constraint.relation.CopyFrom(self.constraint_type.encode())
+        elif self.constraint_type.type == ConstraintTypes.WITHIN:
+            constraint.range_.CopyFrom(self.constraint_type.encode())
+        elif (
+            self.constraint_type.type == ConstraintTypes.IN
+            or self.constraint_type.type == ConstraintTypes.NOT_IN
+        ):
+            constraint.set_.CopyFrom(self.constraint_type.encode())
+        elif self.constraint_type.type == ConstraintTypes.DISTANCE:
+            constraint.distance.CopyFrom(self.constraint_type.encode())
+        else:
+            raise ValueError(
+                f"Incorrect constraint type. Expected a ConstraintTypes. Found {self.constraint_type.type}."
+            )
+        return constraint
+
+    @classmethod
+    def decode(cls, constraint_protobuf_object) -> "Constraint":
+        """
+        Decode a protocol buffer object that corresponds with this class into an instance of this class.
+
+        :param constraint_protobuf_object: the protocol buffer object corresponding with this class.
+        :return: A new instance of this class matching the protocol buffer object
+        """
+        constraint_case = constraint_protobuf_object.WhichOneof("constraint")
+        if constraint_case == "relation":
```

Comment:
Magic values should be declared as constants. Here you can declare as class level constant `RELATION = "relation"` and then access as `self.RELATION`. That way we increase consistency across code and enable reuse.

Applies throughout the file

Reply:
Done